Feature: allow attributes for <fieldset> wrapper

mimmi20 opened the pull request:
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | no

### Description

With this PR Attributes will be allowed an the fieldset wrapper for radio- oder multicheckbox-Elements. This PR is related to Issue #62.

mimmi20 committed:
allow attributes for <fieldset> wrapper

Signed-off-by: Thomas Müller <mimmi20@live.de>

mimmi20 committed:
allow attributes for <fieldset> wrapper

Signed-off-by: Thomas Müller <mimmi20@live.de>

mimmi20 committed:
remove import

Signed-off-by: Thomas Müller <mimmi20@live.de>

mimmi20 committed:
remove import

Signed-off-by: Thomas Müller <mimmi20@live.de>

froschdesign commented on src/View/Helper/FormRow.php:
```diff
@@ -186,8 +187,24 @@ public function render(ElementInterface $element, ?string $labelPosition = null)
+
                 $markup = sprintf(
-                    '<fieldset><legend>%s</legend>%s</fieldset>',
+                    '<fieldset%s><legend%s>%s</legend>%s</fieldset>',
```
The problem is that the markup (`fieldset` and `legend`) [is still hard-coded](https://github.com/laminas/laminas-form/issues/62#issue-591668102). The idea was therefore to use the wrapper functionality from the `FormCollection` helper and extend it. This would allow the same handling between the different helpers.
The `legend_attributes` option which is introduced here also specifies that it is always a `legend` element.

froschdesign replied:
In addition, I ask myself if the handling is not too cumbersome for the user, because the same options would have to be set for each MultiCheckbox or Radio element every time. 🤔 
(Related to https://github.com/laminas/laminas-form/issues/62#issuecomment-618213926)

mimmi20 committed:
update helper

Signed-off-by: Thomas Müller <mimmi20@live.de>

mimmi20 committed:
updates

mimmi20 committed:
reduce complexity

Signed-off-by: Thomas Müller <mimmi20@live.de>

gsteel committed:
Improve type inference for FormElementManager::get()

Signed-off-by: George Steel <george@net-glue.co.uk>

mimmi20 committed:
revert change

Signed-off-by: Thomas Müller <mimmi20@live.de>

mimmi20 committed:
revert changes

Signed-off-by: Thomas Müller <mimmi20@live.de>

mimmi20 committed:
revert issue

Signed-off-by: Thomas Müller <mimmi20@live.de>

Xerkus committed:
Merge pull request #259 from mimmi20/feature-reduce-complexity

Reducing the complexity of view helpers

gsteel committed:
Form element manager always returns an element interface or throws an exception.

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Baseline issue with `FormElementManager::get()` related to inheritance

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Prefer `@return` over `@psalm-return`

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Assert that a class string not implementing ElementInterface is treated the same as a string literal

Signed-off-by: George Steel <george@net-glue.co.uk>

Xerkus committed:
Merge pull request #260 from gsteel/form-element-manager-template

Improve type inference for FormElementManager::get()

Xerkus committed:
Update docs-build workflow

Signed-off-by: Aleksei Khudiakov <aleksey@xerkus.pro>

gsteel committed:
Add failing test case for integer element labels

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Expand types for `translateLabel()` argument to preserve legacy BC.

String is the correct argument type, but it is not possible to implement in the 3.x series

Signed-off-by: George Steel <george@net-glue.co.uk>

Xerkus committed:
Merge pull request #261 from gsteel/Integer-Label-BC-Break

Add failing test case for integer element labels

gsteel committed:
Merge pull request #262 from laminas/3.19.x-merge-up-into-3.20.x_MkCXLtZO

Merge release 3.19.1 into 3.20.x
